sc_demographics.R

```diff
@@ -0,0 +1,188 @@
+
+#####################################################
+# Convert Social Care Demographics
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description -
+#####################################################
+
+
+## output ##
+output <- haven::read_sav(file = "/conf/hscdiip/SLF_Extracts/Social_care/sc_demographics_lookup_Dec_2021.zsav")
+
+
+
+## load packages ##
+library(readr)
+library(odbc)
+library(dplyr)
+library(stringr)
+library(dbplyr)
+library(phsmethods)
+
+
+
+## data ##
+
+######################################################
+# set-up conection to platform
+db_connection <- odbc::dbConnect(
+  odbc::odbc(),
+  dsn = "DVPROD",
+  uid = Sys.getenv("USER"),
+  pwd = rstudioapi::askForPassword("password")
+)
+###################################################
+
+# read in data - social care 2 demographic
+sc_demog <- tbl(db_connection, in_schema("social_care_2", "demographic")) %>%
+  select(
+    latest_record_flag, extract_date, sending_location, social_care_id, upi,
+    chi_upi, submitted_postcode, chi_postcode, submitted_date_of_birth,
+    chi_date_of_birth, submitted_gender, chi_gender_code
+  ) %>%
+  collect()
+
+# variable types
+sc_demog <- sc_demog %>%
+  mutate(
+    submitted_gender = as.numeric(submitted_gender),
+    chi_gender_code = as.numeric(chi_gender_code)
+  )
+
+
+## clean up data ##
+sc_demog <- sc_demog %>%
+  mutate(
+    # use chi if upi is NA
+    upi = coalesce(upi, chi_upi),
+    # check gender code - replace code 99 with 9
+    submitted_gender = replace(submitted_gender, submitted_gender == 99, 9)
+  ) %>%
+  mutate(
+    # use chi gender if avaliable
+    gender = if_else(is.na(chi_gender_code) | chi_gender_code == 9, submitted_gender, chi_gender_code),
+    # use chi dob if avaliable
+    dob = coalesce(chi_date_of_birth, submitted_date_of_birth)
+  )
```

Comment:
We can probably throw a few more of these 'bits' into the SQL by moving the `collect()` down the chain. You might find that something fails because `dbplyr` isn't able to translate the command properly but if it works it should just make the whole script faster as more work is being done in the database rather than transferring the data than working on the R server.

Just an example, I don't know that this will nessecarily work, you could also keep it split, writing back to objects instead of passing on with the pipe. I would recommend making a distinction which contains actual data `sc_demog` with any object which contains the unexecuted query e.g. `sc_demog_query`.
```suggestion
sc_demog <- tbl(db_connection, in_schema("social_care_2", "demographic")) %>%
  select(
    latest_record_flag, extract_date, sending_location, social_care_id, upi,
    chi_upi, submitted_postcode, chi_postcode, submitted_date_of_birth,
    chi_date_of_birth, submitted_gender, chi_gender_code
  ) %>%
# variable types
  mutate(
    submitted_gender = as.numeric(submitted_gender),
    chi_gender_code = as.numeric(chi_gender_code)
  ) %>%
  mutate(
    # use chi if upi is NA
    upi = coalesce(upi, chi_upi),
    # check gender code - replace code 99 with 9
    submitted_gender = replace(submitted_gender, submitted_gender == 99, 9)
  ) %>%
  mutate(
    # use chi gender if avaliable
    gender = if_else(is.na(chi_gender_code) | chi_gender_code == 9, submitted_gender, chi_gender_code),
    # use chi dob if avaliable
    dob = coalesce(chi_date_of_birth, submitted_date_of_birth)
  ) %>% 
    collect()
```

Reply:
`dbplyr` failed this addition